eZ/Publish/API/Repository/Tests/PermissionResolverTest.php

```diff
@@ -249,6 +249,84 @@ public function testHasAccessLimited()
         $this->assertNotEmpty($permissionSets);
     }
 
+    /**
+     * Test for the hasAccess() method.
+     * If policy by design has not any limitation then a user with this policy should have the access.
+     *
+     * @covers       \eZ\Publish\API\Repository\PermissionResolver::hasAccess()
+     *
+     * @dataProvider getDataForTestHasAccessWithLimitations
+     *
+     * @param \eZ\Publish\API\Repository\Values\User\Limitation $limitation
+     * @param $module
+     * @param $function
+     * @param $expectedResult
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\LimitationValidationException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    public function testHasAccessWithLimitations(
+        Limitation $limitation,
+        $module,
+        $function,
+        $expectedResult
+    ): void {
+        $repository = $this->getRepository();
+
+        /* BEGIN: Use Case */
+        $role = $this->createRoleWithPolicies(
+            'role_' . __FUNCTION__,
+            [
+                ['module' => $module, 'function' => $function],
+            ]
+        );
+        $roleService = $repository->getRoleService();
+        $userService = $repository->getUserService();
+        // create user in root user group to avoid overlapping of existing policies and limitations
+        $user = $this->createUser('user', 'John', 'Doe', $userService->loadUserGroup(4));
+        $roleLimitation = $limitation instanceof Limitation\RoleLimitation ? $limitation : null;
```

Comment:
AFAICS both provided Limitations are instances of `Limitation\RoleLimitation`, so this is not needed. 

I'm assuming you wanted to provide a possibility to extend this provider in the future, but this breaks Liskov substitution principle a bit.

Please make 1st argument of this method as `RoleLimitation $roleLimitation` and remove this line. 

If we would want to add another type of limitation, which behaves differently, then we'd have to do it using another argument.

Reply:
I was asking @mikadamczyk about it too, and he pointed me onto \eZ\Publish\API\Repository\Tests\PermissionResolverTest::testCanUserWithLimitations
is this case somewhat different from that?

Reply:
@alongosz fixed

Reply:
> I was asking @mikadamczyk about it too, and he pointed me onto \eZ\Publish\API\Repository\Tests\PermissionResolverTest::testCanUserWithLimitations
is this case somewhat different from that?

@ViniTou no, it's me being a noob :) I probably need to fix it as well... when "there is free time" :joy: